src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -323,28 +325,26 @@ public TopicPartitionChannel(
    * @param kafkaSinkRecord input record from Kafka
    */
   public void insertRecordToBuffer(SinkRecord kafkaSinkRecord) {
-    final long currentOffsetPersistedInSnowflake = this.offsetPersistedInSnowflake.get();
-    final long currentProcessedOffset = this.processedOffset.get();
```

Comment:
Crux of the issue: shouldIgnoreAddingRecordToBuffer modifies processedOffset, but this currentProcessedOffset ignores the change